jms/1.8/modules/ROOT/pages/jms-troubleshooting.adoc

```diff
@@ -78,6 +78,130 @@ To resolve this scenario:
 
 Implement a fallback solution by managing the life cycle of containers to detect when the primary node fails and starts a new node.
 
+[[prefetch-issues]]
+== Troubleshoot Prefetch Issues
+
+A known issue is that a JMS driver or broker prefetch configuration combined with MANUAL ACK and concurrent consumers or consuming more than one message per flow lead to an unexpected ordering of the consumed messages.
+
+JMS brokers implement prefetch policies for efficient usage of network resources using a push model to dispatch messages to consumers, ensuring that a consumer always has a local buffer of messages ready to process.
+When the JMS driver or broker has a prefetch size configuration, each consumer holds a buffer of messages ready to consume. The number of messages in the buffer depends on the broker implementation and custom parameterization.
+
+The JMS Connector *Consume* operation consumes one message at a time using a per operation consumer, which means that each operation holds a buffer of messages, but the operation consumes only the first message in the buffer by design. Which results in an unexpected behavior in the order in which messages are consumed. Consider the following scenarios:
+
+=== Concurrent Flows
+
+Conditions +
+
+* The same flow runs concurrently in two different threads of execution (thead A and thread B).
+* The JMS driver/broker prefetch size is configured to 10.
+* There are 100 messages on the queue.
+
+Flow iteration 1: +
+
+* Thread A: consumer A is created and in its local buffer there are 10 messages ready to process pushed by the broker (messages 1 to 10).
```

Comment:
Maybe rewrite like this:

consumer A is created, and there are 10 messages in its local buffer that are ready to process and be pushed by the broker